File: 515-find-largest-value-in-each-tree-row/515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include <limits.h>

class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        if ( !root )
            return {};
        
        std::queue<TreeNode *> q;
        vector<int> result;
        
        q.push(root);
        while ( !q.empty() )  {
            int lSize = q.size();
            int lMax = INT_MIN;
            
            for (int i = 0; i < lSize; i++) {
                TreeNode *tmp = q.front();
                q.pop();
                
                if ( lMax < tmp->val )
                    lMax = tmp->val;
                
                if ( tmp->left )
                    q.push ( tmp->left );
                if ( tmp->right )
                    q.push( tmp->right );
            }         
            result.push_back(lMax);            
        }
        return result;     
    }
};

File: 515-find-largest-value-in-each-tree-row/515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if root is None:
            return []
        result = []
        
        q = collections.deque([root])
        while len(q) > 0:
            numNode = len(q)
            rowMax = float("-inf")
            for _ in range(numNode):
                node = q.popleft()
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
                #if rowMax < node.val:
                rowMax = max(rowMax, node.val)
            result.append(rowMax)
        
        return result
        